File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    max_w = 0
    max_h = 0
    
    for w, h in sizes:
        w, h = max(w, h), min(w, h) 
        max_w = max(max_w, w)
        max_h = max(max_h, h)
    
    return max_w * max_h
# 명함은 회전이 가능함 -> (w,h), (h,w) 모두 가능 
# 어떤 명함은 가로가 길고, 어떤 명함은 세로가 길면 전체 크기를 정할때 기준이 깨짐 
# 그래서 가로로 기준을 잡음  line6

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cpp
#include <string>
#include <vector>

#include <algorithm>

using namespace std;

int solution(vector<vector<int>> sizes) {
    int maxW=0;
    int maxH=0;
    
    for (int i=0;i<sizes.size();i++){
        int W =max(sizes[i][0], sizes[i][1]);
        int H =min(sizes[i][0], sizes[i][1]);
        
        maxW=max(maxW,W);
        maxH=max(maxH,H);
        
    }
    
    return maxW*maxH;
        
        
}

// 모든 명함을 “긴 쪽 = 가로”, “짧은 쪽 = 세로”로 통일
    
